torch/utils/data/dataloader_experimental.py

```diff
@@ -0,0 +1,89 @@
+
+import functools
+
+import torch.utils.data.backward_compatibility
+from torch.utils.data import DataLoader, IterDataPipe
+from torch.utils.data.datapipes.iter import IterableAsDataPipe
+
+class DataLoader2:
+    def __new__(cls,
+                dataset,
+                batch_size=1,
+                shuffle=False,
+                sampler=None,
+                batch_sampler=None,
+                num_workers=0,
+                collate_fn=None,
+                pin_memory=False,
+                drop_last=False,
+                timeout=0,
+                worker_init_fn=None,
+                *,
+                prefetch_factor=2,
+                persistent_workers=False,
+                batch_outside_worker=False):
+        if isinstance(dataset, IterDataPipe):
+            datapipe = dataset
+            if batch_sampler is not None:
+                raise Exception(
+                    'batch_sampler is not yet supported for DataPipes')
+            if sampler is not None:
+                raise Exception(
+                    'sampler is not yet supported for DataPipes')
+            if shuffle:
+                datapipe = datapipe.shuffle()
+            if batch_outside_worker and pin_memory:
+                raise Exception(
+                    'pin_memory is not yet compatible with batch_outside_worker')
+            if not batch_outside_worker:
+                if batch_size is not None:
+                    datapipe = datapipe.batch(batch_size, drop_last=drop_last)
+                    if collate_fn is None:
+                        collate_fn = torch.utils.data._utils.collate.default_collate
+
```

Comment:
Do we need to add a warning when the pipeline doesn't have `is_shardable` DataPipe with multiple workers?